jerry-core/parser/js/js-scanner-util.c

```diff
@@ -455,6 +455,17 @@ scanner_pop_literal_pool (parser_context_t *context_p, /**< context */
 {
   scanner_literal_pool_t *literal_pool_p = scanner_context_p->active_literal_pool_p;
   scanner_literal_pool_t *prev_literal_pool_p = literal_pool_p->prev_p;
+
+  if (literal_pool_p->source_p == NULL)
+  {
+    JERRY_ASSERT (literal_pool_p->status_flags & SCANNER_LITERAL_POOL_FUNCTION);
+    scanner_context_p->active_literal_pool_p = literal_pool_p->prev_p;
+
+    parser_list_free (&literal_pool_p->literal_pool);
```

Comment:
I think if source_p is NULL, there cannot be any item in the literal pool. Just add an assert.

Reply:
True.